apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/Logging/GoogleLoggerScopeTest.cs

```diff
@@ -0,0 +1,101 @@
+            Assert.Null(GoogleLoggerScope.Current);
+            using (new GoogleLoggerScope("grandparent"))
+            {
+                Assert.Equal("grandparent => ", GoogleLoggerScope.Current.ToString());
+                using (new GoogleLoggerScope("parent"))
+                {
+                    Assert.Equal("grandparent => parent => ", GoogleLoggerScope.Current.ToString());
+                    using (new GoogleLoggerScope("child"))
+                    {
+                        Assert.Equal("grandparent => parent => child => ", GoogleLoggerScope.Current.ToString());
+                    }
+                }
+                Assert.Equal("grandparent => ", GoogleLoggerScope.Current.ToString());
+                using (new GoogleLoggerScope("other_parent"))
+                {
+                    Assert.Equal("grandparent => other_parent => ", GoogleLoggerScope.Current.ToString());
+                }
+                Assert.Equal("grandparent => ", GoogleLoggerScope.Current.ToString());
+            }
+            Assert.Null(GoogleLoggerScope.Current);
+        }
+
+        [Fact]
+        public async Task Multiple_Tasks()
+        {
+            Func<string, string, string, Task> func = async (grandparent, parent, child) =>
+            {
+                Assert.Null(GoogleLoggerScope.Current);
+                using (new GoogleLoggerScope(grandparent))
+                {
+                    Assert.Equal($"{ grandparent} => ", GoogleLoggerScope.Current.ToString());
+                    await Task.Yield();
+                    using (new GoogleLoggerScope(parent))
+                    {
+                        Assert.Equal($"{grandparent} => {parent} => ", GoogleLoggerScope.Current.ToString());
+                        await Task.Yield();
+                        using (new GoogleLoggerScope(child))
+                        {
+                            Assert.Equal($"{grandparent} => {parent} => {child} => ", GoogleLoggerScope.Current.ToString());
+                            await Task.Yield();
+                        }
+                        await Task.Yield();
+                        Assert.Equal($"{grandparent} => {parent} => ", GoogleLoggerScope.Current.ToString());
+                    }
+                    await Task.Yield();
+                    Assert.Equal($"{ grandparent} => ", GoogleLoggerScope.Current.ToString());
+                }
+                Assert.Null(GoogleLoggerScope.Current);
+            };
+
+            await Task.WhenAll(
+                    Task.Run(() => func("grandparent-one", "parent-one", "child-one")),
+                    Task.Run(() => func("grandparent-two", "parent-two", "child-two")),
+                    Task.Run(() => func("grandparent-three", "parent-three", "child-three")),
+                    Task.Run(() => func("grandparent-four", "parent-four", "child-four")),
+                    Task.Run(() => func("grandparent-five", "parent-five", "child-five")));
+        }
+    }
```

Comment:
You should probably also have tests where you open a scope and then within the scope start a task and/or call an async method where you open a child scope and make sure things work as expected. You may want to use the async `SimpleManagedTracerTest.MultipleSpans_*` tests as a guide.